### ISSUE TITLE
Performance RX570 Sapphire Nitro+

### ISSUE THREAD

**Issue by JeanMichelleJarre** [cheneyveron/hackintosh-clover-tachi-x370-1700x-rx570#5]
Hi! I wanted to ask you a question regarding the graphics card. I am using your EFI as I commented in the post I opened a few days ago and I find it curious one thing.
My hackintosh is going very well, really, but I'm surprised that some of the tests I do on the graphics card, some are very good and others very very bad, especially those related to OpenGL / OpenCL. I pass some results of the different benchmark programs:

Geekbench 4 (OpenCL) = 41000 pts (Windows 130,000)
Unigine Valley (ExtremeHD) = 24fps average (Windows 49fps)
Luxmark = 12800 (Very Good)
BruceX 5K FCPX = 23,5s (I think good)
Cinebench R15 (OpenGL) = 72fps (Windows 128fps)

I don't understand how there can be so much difference. The card works perfect in some tests but in others 3 or 4 times lower compared to Windows.
Do you have any idea why it can be? I have read a lot to other colleagues and there are many who are the same, especially with the Rx570 and with the Rx580 in AMD Hackintosh.

I know that OpenGL is in extinction on Mac and that Metal is going to be used but I would like to know why in Hackintosh Intel + Rx570 the card gives very good results and in Hackintosh AMD + Rx570 not, and if something could be done to fix it.

Thank you. Greetings.

**Comment by cheneyveron** [comment]
1. My Taichi X370 with 1700x performs get 44750 in OpenCL in 10.14.4. [Check it here](https://browser.geekbench.com/v4/compute/3970391).
2. I can confirm that 1700x offers no acceleration in Final Cut Pro X.

So, you can see there're still problems with both Virtualization and Kernel Patch solution for AMD Hackintosh which most of them are hard to solve.

For Intel platform such as X79 / X99 / X299, the processors and platform chipsets are rather similar to the processors that Macs use so i'd willing to dig into some problems since it's almost certain that they can be solved. But digging into AMD problems may not lead to a result and worth the time.

**Comment by JeanMichelleJarre** [comment]
Hi!
1.- Yes, OPENGL and OPENCL are not going well but METAL is amazing.
   I updated Lilu.kext and WhatEverGreen.kext but it's still the same ...

2.- I do have acceleration in FCPX with almost the same config (Ryzen 2700). I can work perfectly. Try putting **iMacPro1,1 in SMBIOS from Clover Configurator**. By putting that SMBIOS I get this from the image:

![Captura de pantalla 2019-08-23 a las 17 22 17](https://user-images.githubusercontent.com/54317354/63603847-b27b4700-c5ca-11e9-9806-fc793971ebdd.png)

(Image **Videoproc.app**)

I figured it would have to do with AMD, not with the graphics card, and fixing that (if possible) is very expensive in time. Anyway, even if the graphics card does not go 100%, Mojave on my PC with your EFI is excellent! And I think that changing the SMBIOS to iMacPro1,1 still works faster...

I will be attentive in case you make more changes and update for our motherboard and graphics and I will try to contribute my little grain of sand based always on your great work. Thank you!

**Comment by JeanMichelleJarre** [comment]
Check out this: O_O

https://forum.amd-osx.com/viewtopic.php?f=55&t=8927&hilit=guide

**Comment by cheneyveron** [comment]
I sold my 1700x lol have u tried using 15_16h?

**Comment by JeanMichelleJarre** [comment]
Hi there! Well, this morning I started to install everything again to test if it was true that with 10.4.6 many problems with the graphics were solved.

I created a USB with the instructions of https://vanilla.amd-osx.com/. Everything has gone perfect and started at the first. I have put several kext of my board in the USB EFI and everything went great.

Everything works and the best thing is that the graphics card has actually been fixed. I notice the system much more stable and fluid with this 10.4.6. I have not had any kind of error. The mistakes they gave me with the Adobe suite, no longer gives me. I have been able to install Photoshop and Illustrator 2019 and, together with the solution I put above, they work perfectly!

The best of all? That I have hardware acceleration, of course, and the results of the graphics card in OpenGL and OpenCL:

Geekbench 4 (OpenCL) = 115000 pts (Windows 130,000)
Unigine Valley (ExtremeHD) = 40 fps average (Windows 49fps)
Luxmark = 13090 (Very Good)
BruceX 5K FCPX = 21 s (Very good)
Cinebench R15 (OpenGL) = 115 fps (Windows 128fps)

It has cost work but I already have it working great! =)
